pkg/storageops/gce/gce.go

```diff
@@ -597,7 +620,10 @@ func (s *gceOps) waitForAttach(
 	devicePath, err := task.DoRetryWithTimeout(
 		func() (interface{}, bool, error) {
 			devicePath, err := s.DevicePath(disk.Name)
-			if err != nil {
+			if se, ok := err.(*storageops.StorageError); ok &&
```

Comment:
In the DevicePath API(), we are not determining whether a volume is attached on remote node, but inferring it if it the disk shows as attached and the instance does not show any reference for this disk.

From GCE perspective is it safe to assume that, Disk's status changing and the Disk appearing in DesribeInstance output an atomic operation?

Reply:
Discussed offline. This can be improved to just use disk users.